java/src/main/java/org/rocksdb/RocksDB.java

```diff
@@ -2417,6 +2417,92 @@ public List<ByteBufferGetStatus> multiGetByteBuffers(final ReadOptions readOptio
     return results;
   }
 
+  public boolean keyExists(final byte[] key) {
+    return keyExists(key, 0, key.length);
+  }
+  public boolean keyExists(final byte[] key, final int offset, final int len) {
+    return keyExists(null, null, key, offset, len);
+  }
+  public boolean keyExists(final ColumnFamilyHandle columnFamilyHandle, final byte[] key) {
+    return keyExists(columnFamilyHandle, key, 0, key.length);
+  }
+  public boolean keyExists(final ColumnFamilyHandle columnFamilyHandle, final byte[] key,
+      final int offset, final int len) {
+    return keyExists(columnFamilyHandle, null, key, offset, len);
+  }
+
+  public boolean keyExists(final ReadOptions readOptions, final byte[] key) {
+    return keyExists(readOptions, key, 0, key.length);
+  }
+  public boolean keyExists(
+      final ReadOptions readOptions, final byte[] key, final int offset, final int len) {
+    return keyExists(null, readOptions, key, offset, len);
+  }
+
+  public boolean keyExists(final ColumnFamilyHandle columnFamilyHandle,
+      final ReadOptions readOptions, final byte[] key) {
+    return keyExists(columnFamilyHandle, readOptions, key, 0, key.length);
+  }
+
```

Comment:
My text for the comment would be this:

   * Check if a key exists in the database.
   * This method is not as lightweight as {@code keyMayExist} but it gives a 100% guarantee
   * of a correct result, whether the key exists or not.
   *
   * Internally it checks if the key may exist and then double checks with read operation
   * that confirms the key exists. This deals with the case where {@code keyMayExist} may return
   * a false positive.
   * 
   * The code crosses the Java/JNI boundary only once.